OrbitCore/Callstack.h

```diff
@@ -7,20 +7,24 @@
 
 #include <xxhash.h>
 
-#include <string>
 #include <vector>
 
 #include "CallstackTypes.h"
 
 struct CallStack {
   CallStack() = default;
-  inline CallstackID Hash() {
-    if (m_Hash != 0) return m_Hash;
-    m_Hash = XXH64(m_Data.data(), m_Data.size() * sizeof(uint64_t), 0xca1157ac);
-    return m_Hash;
-  }
+  explicit CallStack(std::vector<uint64_t>&& addresses) {
+    data = std::move(addresses);
+    hash = XXH64(data.data(), data.size() * sizeof(uint64_t), 0xca1157ac);
+  };
 
-  CallstackID m_Hash = 0;
-  std::vector<uint64_t> m_Data;
+  CallstackID GetHash() const { return hash; }
+  uint64_t GetData(size_t id) const { return data.at(id); }
+  const std::vector<uint64_t>& GetData() const { return data; };
+  size_t GetDataSize() const { return data.size(); }
+
+ private:
+  CallstackID hash = 0;
+  std::vector<uint64_t> data;
```

Comment:
`frames_`? `pcs_`? `ips_`? Use the underscore at the end.

Reply:
If it's struct we shouldn't use underscore at the end

Reply:
That's because structs typically only have public members. These are private members, hence why this should be a class and why these should have the underscore.